src/query.ts

```diff
@@ -312,19 +315,22 @@ export function stringifyEncodingQueryFieldDef(encQ: EncodingQuery): string {
   // TODO: convert this chunk into a loop of scale, axis, legend
   if (encQ.scale && !isEnumSpec(encQ.scale)) {
       if (encQ.scale && !isEnumSpec(encQ.scale)) {
-      var scaleParams = {};
-
-      if (encQ.scale['type']) {
-        scaleParams = {type: encQ.scale['type']};
-      }
-      // TODO: push other scale properties to scaleParams.
-
-      if (keys(scaleParams).length > 0) {
-        params.push({
-          key: 'scale',
-          value: JSON.stringify(scaleParams)
-        });
-      }
+
+        const scaleParamsArr = getNestedEncodingPropertyChild('scale');
+        var scaleParams = scaleParamsArr.reduce((scaleParamsObj, param) => {
+            console.log(param);
+            if (encQ.scale[param as string]) {
```

Comment:
Stylistically, I don't think you need any `param as string` here -- it reads quite annoyingly long.  If you really do due to error, make `param: any` above in the method signature.